packages/suite/src/views/settings/SettingsDebug/Passwords.tsx

```diff
@@ -0,0 +1,256 @@
+                    <PasswordEntryPassword>
+                        {!decodedPassword ? (
+                            <Button onClick={decode} type="button" variant="tertiary">
+                                {inProgress ? '....' : 'decode'}
+                            </Button>
+                        ) : (
+                            <>{decodedPassword}</>
+                        )}
+                    </PasswordEntryPassword>
+                </PasswordEntryBody>
+            </PasswordEntryRow>
+        </>
+    );
+};
+
+export const PasswordManager = () => {
+    // todo: filename is saved only locally. for production grade state of this feature we will of course need to save it into app state.
+    const [fileName, setFileName] = useState('');
+    const [providerConnecting, setProviderConnecting] = useState(false);
+    const [fetchingPasswords, setFetchingPasswords] = useState(false);
+
+    const { device } = useSelector(state => ({
+        metadata: state.metadata,
+        device: state.device.selectedDevice,
+    }));
+
+    const selectedProvider = useSelector(selectSelectedProviderForPasswords);
+
+    const { fetchPasswords, connectProvider, disconnectProvider } = useActions({
+        connectProvider: metadataActions.connectProvider,
+        disconnectProvider: metadataActions.disconnectProvider,
+        fetchPasswords: metadataActions.fetchPasswords,
+    });
+
+    const passwordEntries: PasswordEntry[] = useMemo(() => {
+        if (!fileName || !selectedProvider || !selectedProvider?.data?.[fileName]) {
+            return [];
+        }
+
+        const data = selectedProvider?.data![fileName];
+        if ('entries' in data) {
+            return data.entries as PasswordEntry[];
+        }
+        return [];
+    }, [selectedProvider, fileName]);
+
+    const connect = useCallback(() => {
+        setFileName('');
+        setProviderConnecting(true);
+        TrezorConnect.cipherKeyValue({
+            device: { path: device?.path },
+            override: true,
+            useEmptyPassphrase: true,
+            path: PATH,
+            key: DEFAULT_KEYPHRASE,
+            value: DEFAULT_NONCE,
+            encrypt: true,
+            askOnEncrypt: true,
+            askOnDecrypt: true,
+        })
+            .then(async res => {
+                if (!res.success) {
+                    throw new Error(res.payload.error);
+                }
+                const encryptionKey = Buffer.from(
+                    res.payload.value.substring(
+                        res.payload.value.length / 2,
+                        res.payload.value.length,
+                    ),
+                    'hex',
+                );
+
+                const fileKey = res.payload.value.substring(0, res.payload.value.length / 2);
+                const fname = `${crypto
+                    .createHmac('sha256', fileKey)
+                    .update(FILENAME_MESS)
+                    .digest('hex')}.pswd`;
```

Comment:
I suggest to extract this code to separate function to improve readability. 

```ts
// Probably better naming for this function needs to be find
const parsePayload = (value: string) => {
    const encryptionKey = Buffer.from(value.substring(value.length / 2, value.length), 'hex');

    const fileKey = value.substring(0, value.length / 2);
    const fname = `${crypto
        .createHmac('sha256', fileKey)
        .update(FILENAME_MESS)
        .digest('hex')}.pswd`;

    return { encryptionKey, fileKey, fname };
};
```

Reply:
please not now...